src/csmptlv/csmptlv.c

```diff
@@ -181,21 +182,21 @@ const uint8_t *csmptlv_find(const uint8_t *buf, size_t len, tlvid_t qtlvid, uint
 
 int csmptlv_str2id(const char *str, tlvid_t *ptlvid) {
   int rv;
-  rv = sscanf(str,"e%u.%u",&ptlvid->vendor,&ptlvid->type);
+  rv = sscanf(str,"e%" SCNu32 ".%" SCNu32, &ptlvid->vendor,&ptlvid->type);
```

Comment:
This string parsing is based on the extended TLV format as "**e(vendor-id).(tlv-id)**" - any specific reasoning for updating this?

Reply:
> This string parsing is based on the extended TLV format as "**e(vendor-id).(tlv-id)**" - any specific reasoning for updating this?

Hi, format specifiers depend on the platform. Basic linux is 64 bit, but EFR32 is 32 bit platform. Without this solution, we would see several printf/scanf warnings at compile time.

Reply:
Double checked. This solution is required to solve type mismatch on different platforms. If you try to remove this modification and compile EFR32 project, you will see compiler warnings. It's a very basic example:
%u: 16 bit unsigned integer on EFR32 (32 bit), 32bit unsigned integer on Linux 64 bit (original linux target on modern Ubuntu distort > v16.04)

Anyway in the code the types have explicit length, like uint16_t, uint32_t, so there isn't any arithmetic overflow issue, it's just for different printf() functions (32bit vs 64bit). You can try it with in 32bit compiler to replace the hw requirements. Thank you!

Reply:
Detailed explanation:

I just printed the macros on EFR32 and Linux 64 bit platforms:
![Screenshot 2024-07-19 at 14 14 42](https://github.com/user-attachments/assets/e1e7eb4b-5121-4639-b769-b5451fd4402e)

So, we need to use "lu" on EFR32 and "u" literals on Linux (64 bit) in printf/scanf to represent unsigned 32 bit variable.

Reply:
Thanks for checking and re-confirming on this!